charts/node/templates/statefulset.yaml

```diff
@@ -86,6 +86,7 @@ spec:
           args:
             - -c
             - |
+              trap 'echo -e "Snapshot restoration failed. Checkout the logs for errors.\nRemoving /chain-data/chains/${CHAIN_PATH}/{{ $databasePath }} ..."; rm -rf /chain-data/chains/${CHAIN_PATH}/{{ $databasePath }}' ERR
```

Comment:
can you move this after  this if condition?
`if [ -d "/chain-data/chains/${CHAIN_PATH}/{{ $databasePath }}" ];`

Reply:
for example we update base container to alpine and now `set -eu -o pipefail` option is not supported. This trap may remove working DB.

Reply:
Let's make sure this can never be executed on a node which already has a populated disk.

Reply:
> can you move trap after this if condition?
if [ -d "/chain-data/chains/${CHAIN_PATH}/{{ $databasePath }}" ];

Moving it inside `if` won't make much sense as it will define `trap` instruction only if directory already exists. Any errors in the `else` scope will not be trapped and this is where the main logic of the script is happening.

I can move `trap` right after `set` instruction so that if `set` is using incompatible options `trap` will not execute it's cleanup logic.

The script still checks for chain data directory existence before going any further. So you just need the first `if` to run without errors for `trap` to not be executed:
```
if [ -d "/chain-data/chains/${CHAIN_PATH}/{{ $databasePath }}" ]; then
  echo "Database directory already exists, skipping chain snapshot download"
```

btw, `set -eu -o pipefail` IS supported in Alpine images

Reply:
> Any errors in the else scope will not be trapped and this is where the main logic of the script is happening.

Yes, my proposal was to move it inside else block, sorry for miscommunication.  Also can you do it for `download-relay-chain-snapshot` init container too?